web/src/components/vis/ScorePlot.vue

```diff
@@ -208,111 +208,24 @@ export default {
         height,
       } = this;
 
-      if (!valid) {
-        return '';
-      }
-
-      const fmt = format('.2%');
-      const x = `PC${pcX} (${fmt(pcVariances[0])})`;
-      const y = `PC${pcY} (${fmt(pcVariances[1])})`;
-
-      this.chart.axis.labels({
-        x,
-        y,
-      });
-
-      this.chart.resize({
+      return {
+        pcPoints,
+        pcX,
+        pcY,
+        showEllipses,
+        ellipseVisible,
+        duration,
+        pcVariances,
+        valid,
         width,
         height,
-      });
-
-      const [xData, yData] = pcPoints;
-      const grouped = this.grouped(xData, yData);
-
-      const groups = Object.keys(grouped);
-      const columns = [];
-      const labels = {};
-      const xs = {};
-      groups.forEach((g) => {
-        const xName = `${g}_x`;
-
-        columns.push([xName, ...grouped[g].map(d => d.x)]);
-        columns.push([g, ...grouped[g].map(d => d.y)]);
-
-        labels[g] = [...grouped[g].map(d => d.label)];
-
-        xs[g] = xName;
-      });
-
-      // The following side effect depends on xData and yData in the same way
-      // that the C3 chart itself does, so we disable the linter warning that would
-      // prevent it.
-      //
-      // In general, this is part of the tradeoff of mixing an imperative vis
-      // library like C3 with a reactive framework like Vue. The alternatives are
-      // less attractive than breaking the rules here: for instance, we could set a
-      // series of watchers on the dependencies of this function to explicitly
-      // trigger it, but that presents a serious danger of falling out of sync with
-      // changes in the dependency list.
-      //
-      // eslint-disable-next-line vue/no-side-effects-in-computed-properties
-      this.labels = labels;
```

Comment:
@sgratzl this is the main reason.  Mutations inside of computed properties is discouraged.  Mutations inside watchers is perfectly acceptable.

Reply:
I mean not the ScorePlot but other plots that are using a slightly different approach such as ForceDirectedGraph, ....

Reply:
as far as I can, with this change I separate the data needed by a plot and the rendering. Thus, if one uses by accident `this.xxx` inside the `update` method directly, it won't be tracked.

Reply:
Moreover, in this specific case, why is `this.labels` not a computed property directly? Moreover, this logic will change once #387 is implemented again.

Reply:
That's a fair point. In my experience, location is one of the things I'm most aware of when writing vue code.  I think this pattern is more idiomatic Vue because even a novice Vue user would not expect a method to get called in the same way a computed prop is.

I'm not sure if that makes sense.

Reply:
@sgratzl I hadn't realized you'd put a very similar mechanism into the `axisPlot` mixin. After the November release, I think it would make a lot of sense to abstract that functionality into its own mixin, and use it in *any* component that relies on imperative, non-Vue mechanisms to fulfill their role.

I will close this PR and open an issue to tackle this later. Thanks for weighing in on this.

Reply:
@subdavis pointed out to me off-channel that there may still be some merit to this PR. I'll leave it open but we can consider the issues more fully after the November release.